src/socket-proxy/socket-proxyd.c

```diff
@@ -297,11 +300,89 @@ static int connection_enable_event_sources(Connection *c) {
         return 0;
 }
 
+static void send_proxy_protocol_v1(Connection *c) {
+        _cleanup_free_ char *s = NULL;
+        int r, s_len;
+
+        union sockaddr_union local_addr, remote_addr;
+        socklen_t addr_len;
+
+        r = sd_is_socket(c->server_fd, AF_UNSPEC, SOCK_STREAM, 0);
+        if (r < 0) {
+                log_warning_errno(errno, "Failed to issue SO_TYPE: %m");
+                goto unknown;
+        }
+        if (r == 0) {
+                log_warning("Only TCP is supported by the PROXY protocol");
+                goto unknown;
+        }
+
+        addr_len = sizeof(local_addr);
+        if (getsockname(c->server_fd, (struct sockaddr*) &local_addr, &addr_len) < 0) {
+                log_warning_errno(errno, "Failed to get local address (getsockname)");
+                goto unknown;
+        }
+
+        addr_len = sizeof(remote_addr);
+        if (getpeername(c->server_fd, (struct sockaddr*) &remote_addr, &addr_len) < 0) {
+                log_warning_errno(errno, "Failed to get remote address (getpeername)");
+                goto unknown;
+        }
+
+        switch (remote_addr.sa.sa_family) {
+        case AF_INET:
+                s_len = asprintf(&s, "PROXY TCP4 %s %s %hu %hu\r\n",
+                        IN4_ADDR_TO_STRING(&(remote_addr.in.sin_addr)),
+                        IN4_ADDR_TO_STRING(&(local_addr.in.sin_addr)),
+                        be16toh(remote_addr.in.sin_port),
+                        be16toh(local_addr.in.sin_port)
+                );
```

Comment:
Spurious indentation. Please align with `(`
```c
s_len = asprintf(&s, "PROXY TCP4 %s %s %hu %hu\r\n",
                 IN4_ADDR_TO_STRING(&(remote_addr.in.sin_addr)),
                 IN4_ADDR_TO_STRING(&(local_addr.in.sin_addr)),
                 be16toh(remote_addr.in.sin_port),
                 be16toh(local_addr.in.sin_port));
```
Also, we typically not use `h` modifier. Please simply use `%u`.